sdk/httputil/http.go

```diff
@@ -0,0 +1,64 @@
+package httputil
+
+import (
+	"crypto/tls"
+	"net/http"
+	"time"
+)
+
+const (
+	defaultTimeout = 120 * time.Second
+	// defaults to match DefaultTransport - defined to satisfy lint
+	maxIdleConns          = 100
+	idleConnTimeout       = 90 * time.Second
+	tLSHandshakeTimeout   = 10 * time.Second
+	expectContinueTimeout = 1 * time.Second
+)
+
+var preventRedirectCheck = func(_ *http.Request, _ []*http.Request) error {
+	return http.ErrUseLastResponse // Prevent following redirects
+}
+
+var safeHTTPClient = &http.Client{
+	Transport:     http.DefaultTransport,
+	Timeout:       defaultTimeout,
+	CheckRedirect: preventRedirectCheck,
+}
+
+// SafeHTTPClient returns a default http client which has sensible timeouts, won't follow redirects, and enables idle
+// connection pooling.
+func SafeHTTPClient() *http.Client {
+	return safeHTTPClient
+}
+
+// SafeHTTPClientWithTLSConfig returns a http client which has sensible timeouts, won't follow redirects, and if
+// specified a http.Transport with the tls.Config provided.
+func SafeHTTPClientWithTLSConfig(cfg *tls.Config) *http.Client {
+	if cfg == nil {
+		return safeHTTPClient
+	}
+	return SafeHTTPClientWithTransport(&http.Transport{
+		TLSClientConfig: cfg,
+		// config below matches DefaultTransport
+		Proxy:                 http.ProxyFromEnvironment,
+		ForceAttemptHTTP2:     true,
+		MaxIdleConns:          maxIdleConns,
+		IdleConnTimeout:       idleConnTimeout,
+		TLSHandshakeTimeout:   tLSHandshakeTimeout,
+		ExpectContinueTimeout: expectContinueTimeout,
+	})
```

Comment:
We might want to expose something in the sdk which will allow developers to change these defaults (since they are defaults). This way we can support a variety of network environments, especially over remote areas with high-latency.

Reply:
I don't think most users will need to concern themselves with this.  But if they have a long running process with spare memory it's conceivable that someone may want to change it.  However I don't believe it should be necessary, and even without precise tuning this will be significantly better than what we have today.

I guess generally I prefer to just keep configuration simple and defaults sane, and only bring in the config complexity when a clear use case is understood.  It's easy to expand, but hard to retract.

That said, no strong preference here since tuning these defaults wont increase risk to our users. If you would like them to be configurable should I just add those values to the `config` struct?  Let me know your preference.